src/Hl7.Fhir.Base/FhirPath/ElementNavFhirExtensions.cs

```diff
@@ -243,5 +255,54 @@ internal static P.Time BoundaryTime(P.Time time, long? precision, int minutes, i
         /// <param name="r"></param>
         /// <returns></returns>
         internal static bool Comparable(P.Quantity l, P.Quantity r) => l.TryCompareTo(r).Success;
+
+        /// <summary>
+        /// When invoked on a single code-valued element, returns true if the code is a member of the given valueset. 
+        /// When invoked on a single concept-valued element, returns true if any code in the concept is a member of the given valueset. 
+        /// When invoked on a single string, returns true if the string is equal to a code in the valueset, so long as the valueset only contains one codesystem. 
+        /// If the valueset in this case contains more than one codesystem, the return value is empty.
+        /// 
+        /// If the valueset cannot be resolved as a uri to a value set, or the input is empty or has more than one value, the return value is empty.
+        /// </summary>
+        /// <param name="input">The input element for the function 'memberOf()'</param>
+        /// <param name="valueset">The valueset</param>
+        /// <param name="ctx">EvaluationContext of the FhirPath compiler</param>
+        /// <returns>See summary</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        internal static bool? MemberOf(ITypedElement input, string valueset, EvaluationContext ctx)
+        {
+            var service = (ctx is FhirEvaluationContext fctx ? fctx.TerminologyService : null)
+                ?? throw new ArgumentNullException("The 'memberOf' function cannot be executed because the FhirEvaluationContext does not include a TerminologyService.");
+
+            ValidateCodeParameters? inParams = new ValidateCodeParameters()
+                        .WithValueSet(valueset);
+
+            inParams = input.InstanceType switch
+            {
+                "code" when input is ScopedNode sn => inParams.WithCode(code: sn.Value as string, context: sn.LocalLocation),
```

Comment:
The `LocalLocation` will look something like `Patient.name[1].family[2]` etc. while according to http://hl7.org/fhir/valueset-operation-validate-code.html, the "context" that we need to pass in is defined as 

> The recommended format for this URI is [Structure Definition URL]#[name or path into structure definition] e.g. http://hl7.org/fhir/StructureDefinition/observation-hspc-height-hspcheight#Observation.interpretation

This is obviously written from the viewpoint of a profile validator, since that would call the service with full information about the profile (Note: even our profile validator does not pass in the canonical as far as I know).
In a "random" fhirpath evaluation on an instance, we would never have a profile uri (at most the canonical to the core definition), so we cannot really align with this requirement.   

Finally, the `LocalPath` has indices for repeating elements, which the context path is not supposed to have. Do we put our head in the sand here for now?

Reply:
Yes, let's just leave this as is and go for it.